data-prepper-core/src/main/java/org/opensearch/dataprepper/sourcecoordination/LeaseBasedSourceCoordinator.java

```diff
@@ -302,7 +300,8 @@ public <S extends T> void saveProgressStateForPartition(final String partitionKe
 
         try {
             sourceCoordinationStore.tryUpdateSourcePartitionItem(itemToUpdate);
-        } catch (final PartitionUpdateException e) {
+        } catch (final Exception e) {
+            LOG.info("Exception while saving state for the partition {}: {}", partitionKey, e.getMessage());
```

Comment:
This should be an `ERROR` log and should include the stack trace since you don't know the underlying cause if you reach this code.

Reply:
Thanks yeah it should be